api/Controllers/DataSetsController.cs

```diff
@@ -55,27 +57,48 @@ public async Task<IActionResult> AutoLabel(AutoLabelConfig config)
     {
         var commits = await _gitCommitService.Get(config.GitRepoId);
         var keywords = await _keywordService.GetByKeywordSetId(config.KeywordSetId);
-        var autoLabeledData = new List<LabeledData>();
+        var concurrentBag = new ConcurrentBag<LabeledData>();
 
-        foreach (var commit in commits)
+        Parallel.ForEach(commits, commit => { concurrentBag.Add(LabelCommit(commit, keywords, config)); });
+
+        ICollection<LabeledData> labeledData = concurrentBag.ToArray();
+        
+        if (config.BalanceOutput)
+        {
+            var usefuls = labeledData.Where(e => e.IsUseful.Equals(true));
```

Comment:
usefulData

Reply:
ik zou het zo laten omdat het anders de indruk wekt dat het om 'zinvolle' data gaat, terwijl de isUseful=FALSE ook wel degelijk zinvol is. Het is alleen een split om ervoor te zorgen dat ik operaties kan uitvoeren op een van beide.